# Pandas Data Science Case Study
This notebook follows along with the "Pandas for Data Science in 20 Minutes" tutorial.
Dataset used: telco_churn.csv

In [1]:
import pandas as pd


In [2]:
df = pd.read_csv("telco_churn.csv")

In [3]:
df.head()

,State,Account length,Area code,International plan,Voice mail plan,Number vmail messages,Total day minutes,Total day calls,Total day charge,Total eve minutes,Total eve calls,Total eve charge,Total night minutes,Total night calls,Total night charge,Total intl minutes,Total intl calls,Total intl charge,Customer service calls,Churn
0,KS,128,415,No,Yes,25,265.1,110.0,45.07,197.4,99.0,16.78,244.7,91.0,11.01,10.0,3,2.70,1.0,False
1,OH,107,415,No,Yes,26,161.6,123.0,27.47,195.5,103.0,16.62,254.4,103.0,11.45,13.7,3,3.70,1.0,False
2,NJ,137,415,No,No,0,243.4,114.0,41.38,121.2,110.0,10.30,162.6,104.0,7.32,12.2,5,3.29,0.0,False
3,OH,84,408,Yes,No,0,299.4,71.0,50.90,61.9,88.0,5.26,196.9,89.0,8.86,6.6,7,1.78,2.0,False
4,OK,75,415,Yes,No,0,166.7,113.0,28.34,148.3,122.0,12.61,186.9,121.0,8.41,10.1,3,2.73,3.0,False


## Dataset Dimensions
The shape function shows the number of rows and columns.

In [4]:
df.shape

(3333, 20)

In [5]:
df.columns

Index(['State', 'Account length', 'Area code', 'International plan',
       'Voice mail plan', 'Number vmail messages', 'Total day minutes',
       'Total day calls', 'Total day charge', 'Total eve minutes',
       'Total eve calls', 'Total eve charge', 'Total night minutes',
       'Total night calls', 'Total night charge', 'Total intl minutes',
       'Total intl calls', 'Total intl charge', 'Customer service calls',
       'Churn'],
      dtype='object')

## Dataset Information
The info() function displays data types and non-null counts for each column.

In [6]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 3333 entries, 0 to 3332
Data columns (total 20 columns):
 #   Column                  Non-Null Count  Dtype  
---  ------                  --------------  -----  
 0   State                   3333 non-null   object 
 1   Account length          3333 non-null   int64  
 2   Area code               3333 non-null   int64  
 3   International plan      3333 non-null   object 
 4   Voice mail plan         3333 non-null   object 
 5   Number vmail messages   3333 non-null   int64  
 6   Total day minutes       3323 non-null   float64
 7   Total day calls         3323 non-null   float64
 8   Total day charge        3315 non-null   float64
 9   Total eve minutes       3324 non-null   float64
 10  Total eve calls         3325 non-null   float64
 11  Total eve charge        3333 non-null   float64
 12  Total night minutes     3333 non-null   float64
 13  Total night calls       3332 non-null   float64
 14  Total night charge      3333 non-null   

## Summary Statistics
The describe() function provides summary statistics for numerical columns.

In [7]:
df.describe()

,Account length,Area code,Number vmail messages,Total day minutes,Total day calls,Total day charge,Total eve minutes,Total eve calls,Total eve charge,Total night minutes,Total night calls,Total night charge,Total intl minutes,Total intl calls,Total intl charge,Customer service calls
count,3333.000000,3333.000000,3333.000000,3323.000000,3323.000000,3315.000000,3324.000000,3325.000000,3333.000000,3333.000000,3332.000000,3333.000000,3333.000000,3333.000000,3328.000000,3328.000000
mean,101.064806,437.182418,8.099010,179.787150,100.456214,30.557831,201.033935,100.110677,17.083540,200.872037,100.115246,9.039325,10.237294,4.479448,2.764588,1.563101
std,39.822106,42.371290,13.688365,54.419625,20.057356,9.255987,50.676652,19.932115,4.310668,50.573847,19.566710,2.275873,2.791840,2.461214,0.754086,1.315870
min,1.000000,408.000000,0.000000,0.000000,0.000000,0.000000,0.000000,0.000000,0.000000,23.200000,33.000000,1.040000,0.000000,0.000000,0.000000,0.000000
25%,74.000000,408.000000,0.000000,143.700000,87.000000,24.420000,166.600000,87.000000,14.160000,167.000000,87.000000,7.520000,8.500000,3.000000,2.300000,1.000000
50%,101.000000,415.000000,0.000000,179.400000,101.000000,30.500000,201.400000,100.000000,17.120000,201.200000,100.000000,9.050000,10.300000,4.000000,2.780000,1.000000
75%,127.000000,510.000000,20.000000,216.500000,114.000000,36.780000,235.325000,114.000000,20.000000,235.300000,113.000000,10.590000,12.100000,6.000000,3.270000,2.000000
max,243.000000,510.000000,51.000000,350.800000,165.000000,59.640000,363.700000,170.000000,30.910000,395.000000,175.000000,17.770000,20.000000,20.000000,5.400000,9.000000


## Missing Values
Check for any missing data in each column.

In [8]:
df.isna().sum()

State                      0
Account length             0
Area code                  0
International plan         0
Voice mail plan            0
Number vmail messages      0
Total day minutes         10
Total day calls           10
Total day charge          18
Total eve minutes          9
Total eve calls            8
Total eve charge           0
Total night minutes        0
Total night calls          1
Total night charge         0
Total intl minutes         0
Total intl calls           0
Total intl charge          5
Customer service calls     5
Churn                      8
dtype: int64

## Churn Distribution
Count how many customers churned versus did not churn.

In [9]:
df["Churn"].value_counts()

Churn
False    2842
True      483
Name: count, dtype: int64

## Average Metrics by Churn Status
Compare average numerical values between churned and non-churned customers.

In [10]:
df.groupby("Churn").mean(numeric_only=True)

,Account length,Area code,Number vmail messages,Total day minutes,Total day calls,Total day charge,Total eve minutes,Total eve calls,Total eve charge,Total night minutes,Total night calls,Total night charge,Total intl minutes,Total intl calls,Total intl charge,Customer service calls
Churn,,,,,,,,,,,,,,,,
False,100.751232,437.003167,8.576003,175.172336,100.282992,29.774446,199.113444,100.073721,16.918455,200.150422,100.073214,9.006847,10.159606,4.527445,2.743596,1.450035
True,102.664596,437.817805,5.115942,206.952599,101.432432,35.167375,212.327178,100.522822,18.054969,205.231677,100.399586,9.235528,10.700000,4.163561,2.889545,2.229814


## Conclusion

The dataset contains 3,333 customers and 20 features. 
Most customers did not churn (2,842 vs 483 who churned). 
On average, customers who churned tend to have higher total day minutes and charges compared to non-churned customers.